### DISCUSSION TITLE
Missing ampersand when passing byref

### DISCUSSION THREAD

**Discussion by radekm** [dotnet/fsharp#14388] [General]
I wanted to load string to `Vector256` so I initially wrote

```fsharp
let s = "Hi, what's your name?"
let span = s.AsSpan()
let r = &MemoryMarshal.GetReference(span)
let v = Vector256.LoadUnsafe<uint16>(Unsafe.As(&r))  // <== & should precede Unsafe.As.
```

It throws `System.NullReferenceException`. Then I realized that `&` is missing before `Unsafe.As`. But now I'm wondering why it even compiles (it seems I'm giving `uint16` where `byref<uint16>` is expected)?

Following doesn't compile as expected:

```fsharp
let r2 : byref<uint16> = &Unsafe.As(&r)
let v2 = Vector256.LoadUnsafe<uint16>(r2)  // <== Doesn't compile (& should precede r2).
```

**Comment by T-Gro** [comment]
As far as I can see in https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzgHwgAcYA7AAgGUBPXAFxgFsA6AJQFdS6BLRmZgJJcYUYpREA3bmBi4AsAChiZKrQYsOXXvyF0o3UrmnylJCjXpM2nHn2YBhCIyLcMI8VCkyTy82qaKgQpudOSM1ABinGDkABQAlOQAvOSK5OnkIeS4yeQARAAS3GjkAO4AFgCGdADkOdQQ7FDkpJV8APx5aRlZuESVFCm4zACCuJT9pAnd6VnNKQBkALJM0NRLlVC4VRjMAOIwdKww2CJkMrF9A/EzmYfkErkAajBgdNAATACsSMwAMhBKgATACqhkqpwAPOwDHQAIxIAB8sTBuAh/DGsQWUHiiQyt0chggbmYAHV9Aw/gYYLEJPtDgAVagkBLMBkQSh6AwAcwSNwUQSBEDCkWiCSAA , the compiler did generate the **ref**  for you in this case.
